rational.c

```diff
@@ -2439,24 +2440,27 @@ parse_rat(const char *s, const char *const e, int strict, int raise)
     if (nexp != ZERO) {
         if (INT_NEGATIVE_P(nexp)) {
             VALUE mul;
-            if (!FIXNUM_P(nexp)) {
```

Comment:
How about `if (!FIXNUM_P(nexp) || RB_FLOAT_TYPE_P(mul = f_expt10(LONG2NUM(-FIX2LONG(nexp)))))`?

Reply:
OK, I take that.  I have no strong opinion.